api/desecapi/views.py

```diff
@@ -323,6 +324,16 @@ def get(self, request, *_):
         try:
             serializer.is_valid(raise_exception=True)
         except ValidationError as e:
+            if isinstance(e.detail, list) and \
+                    any([
+                        any([
+                            getattr(non_field_error, 'code', '') == 'unique'
+                            for non_field_error
+                            in err.get('non_field_errors', [])
+                        ])
+                        for err in e.detail
+                    ]):
+                raise ConcurrencyException from e
```

Comment:
I suppose this happens when a dynDNS update triggers creation of an RRset that was already created by another concurrent request (and the change tracker did not see it yet when it calculated its plan of action). I think a comment explaining that would help here.

However, shouldn't this be part of `serializer.is_valid()`? Why is it on view level?

Reply:
The serializer can't handle this by itself as the `instances` argument -- given to it by the view -- can be outdated due to a concurrent request.